src/modules/esl-utils/dom/events/target.ts

```diff
@@ -34,8 +45,9 @@ export class SyntheticEventTarget implements EventTarget {
     if (typeof type !== 'string') return this.removeEventListener((this.constructor as typeof SyntheticEventTarget).DEFAULT_EVENT, type);
 
     validateEventListenerType(callback);
-    if (!this._listeners[type]) return;
-    this._listeners[type] = this._listeners[type].filter((cb) => cb !== callback);
+    const listeners = this.getEventListeners(type);
+    const index = listeners.indexOf(callback!);
+    if (index >= 0) listeners.splice(index, 1);
```

Comment:
The previous version of the code is considered as more stable because it assumes that array can be corrupted by not unique values